system/lib/mimalloc/src/prim/emscripten/prim.c

```diff
@@ -78,17 +78,10 @@ int _mi_prim_alloc(size_t size, size_t try_alignment, bool commit, bool allow_la
   //       That assumes no one else uses sbrk but us (they could go up,
   //       scribble, and then down), but we could assert on that perhaps.
   *is_zero = false;
-  // emmalloc has some limitations on alignment size.
-  // TODO: Why does mimalloc ask for an align of 4MB? that ends up allocating
-  //       8, which wastes quite a lot for us in wasm. If that is unavoidable,
-  //       we may want to improve emmalloc to support such alignment. See also
-  //       https://github.com/emscripten-core/emscripten/issues/20645
+  // emmalloc has a minimum alignment size.
   #define MIN_EMMALLOC_ALIGN           8
-  #define MAX_EMMALLOC_ALIGN (1024*1024)
   if (try_alignment < MIN_EMMALLOC_ALIGN) {
```

Comment:
Can try_alignment ever be lower than 8 in mimalloc?

Reply:
I think no, since it makes that assumption internally. But we could probably make it do the clamping itself silently, I suppose.

Reply:
Oh sorry, you said mimalloc and I read that as emmalloc. I'm not sure. The initial allocations are definitely 4MB-sized, but there may be other incidental allocations in theory in the codebase somewhere.